### DISCUSSION TITLE
Plans for a native implementation?

### DISCUSSION THREAD

**Discussion by domoritz** [EvgSkv/logica#25] [Ideas]
First off, Logica looks great. I love that modularity and testing are explicitly stated goals of the language. I think it'll make it a great target for analytics. 

Since not everybody is doing data analysis in Python, do you plan on implementing logica in a language that can be easily integrated into other environments? I'm thinking of maybe C, C++, or Rust. You could even reuse that compiler in this Python library.

**Comment by EvgSkv** [comment]
Thank you for your supportive feedback!

Yes, I think Logica should be implemented in C++ in the future for easier integration.
In the short term it could make sense to write a Python server that would provide an API for compilation and maybe even execution of queries, as calling an API could be easier than calling Python code. I understand that this solution is not ideal, as it adds communication overhead, but it could be acceptable for many practical uses.
